trading_calendars/exchange_calendar_asex.py

```diff
@@ -0,0 +1,180 @@
+    'National Holiday 1',
+    month=3,
+    day=25,
+)
+
+OrthodoxGoodFriday = orthodox_easter() - timedelta(2)
+
+OrthodoxEasterMonday = orthodox_easter() + timedelta(1)
+
+LabourDay = european_labour_day()
+
+OrthodoxWhitMonday = orthodox_easter() + timedelta(50)
+
+AssumptionDay = assumption_day()
+
+NationalHoliday2 = Holiday(
+    'National Holiday 2',
+    month=10,
+    day=28,
+)
+
+ChristmasEve = christmas_eve(start_date='2009')
+
+ChristmasDay = christmas()
+
+SecondDayOfChristmas = Holiday(
+    'Second Day of Christmas',
+    month=12,
+    day=26,
+)
+
+
+class ASEXExchangeCalendar(TradingCalendar):
+    """
+    Exchange calendar for the Athens Stock Exchange (ASEX).
+
+    Open Time: 10:00 AM, EEST
+    Close Time: 5:25 PM, EEST
+
+    Regularly-Observed Holidays:
+    - New Year's Day
+    - Epiphany (Jan 6)
+    - Orthodox Ash Monday (48 days before Orthodox Easter Sunday)
+    - National Holiday (Mar 25)
+    - Good Friday
+    - Easter Monday
+    - Orthodox Good Friday
+    - Orthodox Easter Monday
+    - Labour Day (May 1)
+    - Orthodox Whit Monday (50 days after Orthodox Easter Sunday)
+    - Assumption Day (Aug 15)
+    - National Holiday (Oct 28)
+    - Christmas Eve
+    - Christmas Day
+    - Second Day of Christmas (Dec 26)
+
+    Early Closes:
+    - None
+    """
+    name = 'ASEX'
+
+    tz = timezone('Europe/Athens')
+
+    open_times = (
+        (None, time(10, 1)),
+    )
+
+    close_times = (
+        (None, time(17, 00)),
+        (pd.Timestamp('2008-09-29'), time(17, 20))
+    )
+
+    @property
+    def regular_holidays(self):
+        return HolidayCalendar([
+            NewYearsDay,
+            Epiphany,
+            NationalHoliday1,
+            GoodFriday,
+            EasterMonday,
+            LabourDay,
+            AssumptionDay,
+            NationalHoliday2,
+            ChristmasEve,
+            ChristmasDay,
+            SecondDayOfChristmas
+        ])
+
+    @property
+    def adhoc_holidays(self):
+        debt_crisis = pd.date_range('2015-06-29', '2015-07-31', freq='B')
+
+        debt_crisis_holidays = [
+            pd.Timestamp(str(date), tz=UTC) for date in debt_crisis
+        ]
+
+        misc_adhoc_holidays = [
+            # In 2002, market closed for unknown reason
+            pd.Timestamp('2002-05-07', tz=UTC),
+            # In 2004, Assumption Day fell on a sunday, observed on a Friday
+            pd.Timestamp('2004-08-13', tz=UTC),
+            # In 2008, worker strikes closed the market for 2 days in March
+            pd.Timestamp('2008-03-04', tz=UTC),
+            pd.Timestamp('2008-03-05', tz=UTC),
+            # In 2013, May Day strikes closed the market
+            pd.Timestamp('2013-05-07', tz=UTC),
+            # In 2014, New Year's Eve was observed as a holiday
+            pd.Timestamp('2014-12-31', tz=UTC),
+            # In 2016, Labour Day fell on a Sunday, observed on Tuesday
+            pd.Timestamp('2016-05-03', tz=UTC),
+        ]
```

Comment:
Very minor perf note, but since this list and the `debt_crisis` list are just constants, you could just define these at module scope with the other holiday definitions so that the list doesn't have to be recreated even time this method is invoked.

Reply:
I think i'm gonna leave this as is because we have generally just been creating the lists in `adhoc_holidays` and as Scott just pointed out in Slack they only get touched once.